### PR TITLE
atlas-stream: add limit for max-datasources-per-session

### PR DESCRIPTION
**Pull request by jfz**
Limit number of DataSources per session. 
Refactored DataSourceValidator to avoid passing extra parameters everywhere.

### CODE REVIEW

File: atlas-stream/src/main/scala/com/netflix/atlas/stream/DataSourceValidator.scala [hunk 1]
```diff
@@ -40,43 +37,45 @@ object DataSourceValidator {
...
-        addError(id, "id cannot be null", errorMap)
-      } else if (id.isEmpty) {
-        addError(id, "id cannot be empty", errorMap)
-      } else {
-        if (visitedIds.contains(id)) {
-          addError(id, "id cannot be duplicated", errorMap)
+
+    // Validate size limit first
+    if (dataSourceList.size > maxDataSourcesPerSession) {
+      addError("_", s"number of DataSources cannot exceed $maxDataSourcesPerSession", errorMap)
```
Comment (jfz) [inline atlas-stream/src/main/scala/com/netflix/atlas/stream/DataSourceValidator.scala]: quickly fail if over limit without extra info to keep it clear

File: atlas-stream/src/main/scala/com/netflix/atlas/stream/EvalService.scala [hunk 2]
```diff
@@ -203,7 +202,7 @@ object EvalService {
     }
 
     def complete(): Unit = {
-      logger.debug(s"queue complete for $id")
+      logger.info(s"queue complete for: $id")
```
Comment (jfz) [inline atlas-stream/src/main/scala/com/netflix/atlas/stream/EvalService.scala]: worth log it for troubleshooting, one per session only

File: atlas-stream/src/main/resources/application.conf [hunk 3]
```diff
@@ -7,4 +7,7 @@ atlas.akka {
   ]
 }
 
-atlas.stream.eval-service.queue-size = 10000
\ No newline at end of file
+atlas.stream {
+  eval-service.queue-size = 10000
+  max-datasources-per-session = 100
```
Comment (brharrington) [inline atlas-stream/src/main/resources/application.conf]: Will you also be adding a limit for the max overall?

Comment (jfz) [reply atlas-stream/src/main/resources/application.conf]: yes

**Pull request merged**